azblob/list.go

```diff
@@ -13,15 +13,25 @@ import (
 
 // Count counts the number of blobs filtered by the given tags filter
 func (azp *Storer) Count(ctx context.Context, tagsFilter string, opts ...Option) (int64, error) {
-
-	logger.Sugar.Debugf("Count")
-
-	r, err := azp.FilteredList(ctx, tagsFilter, opts...)
-	if err != nil {
-		return 0, err
-	}
-
-	return int64(len(r.Items)), nil
+	log := logger.Sugar.FromContext(ctx)
+	defer log.Close()
+
+	var count int64
+	var m ListMarker
+
+	for {
+		opts = append(opts, WithListMarker(m))
```

Comment:
doesn't this opts list grow infinitely now ?

Reply:
this needs to be 

r, err := azp.FilteredList(ctx, tagsFilter, append(opts,  WithListMarker(m))...)